src/components/Feed/Feed.js

```diff
@@ -132,20 +132,36 @@ export default class Feed extends Component {
     const { channel, user } = this.props;
 
     if (!channel.tags) return alert("Cannot add post to this channel");
-
     const postContent = {
       author: user._id,
       tags: channel.tags,
       content: data.content,
     }
 
     ApiClient.post(`/posts/`, postContent, { authorized: true })
-      .then(() => {
-        this.loadData({ reload: true });
+      .then(response => {
+        if (data.image) {
+          new Promise(function(resolve, reject) {
+            ApiClient.uploadPhoto(
+                `/posts/${response._id}/image`,
+                data.image,
+                'image',
+                {authorized: true, method: 'POST'}
+              ).catch(err => {
+                reject(err);
+              })
+              ;
+            }).then(() => this.loadData({ reload: true }));
```

Comment:
I could be wrong, but I believe on mobile we don't actually do a full refresh. We just assume that the post is successful and then render it locally so that there is no delay between posting it and seeing it in the feed. This way is fine, but rendering it using the local data will speed up how long the process takes.

Reply:
This is not actually changed from the previous picture-less implementation, I just moved where the logic took place. So you may be right but I don't know if it's a blocker for the merge.

Reply:
That's fine, it's not a blocker at all